src/Orleans/CodeGeneration/ConsiderForCodeGenerationAttribute.cs

```diff
@@ -3,18 +3,25 @@ namespace Orleans.CodeGeneration
     using System;
     
     /// <summary>
-    /// Abstract base class for attributes that effect code generation of types.
+    /// The attribute which informs the code generator that code should be generated for this type.
     /// </summary>
-    public abstract class ConsiderForCodeGenerationAttribute : Attribute
```

Comment:
Some have expressed that `[KnownType(typeof(X))]` does not have the most obvious semantics/naming. I believe this is a better attribute name for those purposes.

I believe this is more understandable than the existing attribute, which we can consider marking as obsolete.

I'm open to opinions, though.